mysql-test/lib/mtr_report.pm

```diff
@@ -499,27 +499,39 @@ sub mtr_report_stats ($$$$) {
 
       my $test_result;
 
-      # if a test case has to be retried it should have the result MTR_RES_FAILED in jUnit XML
       if ($test->{'retries'} > 0) {
-        $test_result = "MTR_RES_FAILED";
+        $test_result = "MTR_RES_FAIL";
```

Comment:
@timofeyturenko, @DmitriyKarpovskiy1 , please note that function you are changing is called `mtr_report_stats`. That function is called for *REPORT* only, so *AFTER* all tests produce the result and this function *just* write the statistics about. Look into: `mtr_report_stats($prefix, $fail, $completed, $extra_warnings);` (`mysql-test-run.pl:700)`
In that sense:
Please change this to `$test_result = $test->{'result'}`, no need to check with conditions. This applies iff retried tests are catched in the result. 
Do you really need this status variable `MTR_RES_RETRY_PASSED` in your CI? If it is not enough to know that there are {{retries > 0}} and that test is successful than, I'm ok, but note that this status is the only one used in whole mtr. In that sense just treat this if condition.